packages/nats-runner/src/Consumers.ts

```diff
@@ -9,4 +17,47 @@ export abstract class JetStreamConsumer {
     }
 
     abstract handle(msg: JetStreamMessage<unknown>): Promise<void>;
+
+    abstract onReady(): void;
```

Comment:
you could still let an emtpy function to avoid later tests like `if (typeof consumer.onReady === "function") {`